tach/cache/setup.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import annotations
+
+import os
+import uuid
+from pathlib import Path
+
+
+def resolve_project_dir() -> Path:
+    current_dir = os.getcwd()
+    tach_path = Path(os.path.join(current_dir, "tach.yml"))
+    if not tach_path.exists():
+        raise FileNotFoundError()
+    return Path(current_dir)
+
+
+def resolve_dot_tach() -> Path:
+    project_dir = resolve_project_dir()
+
+    def _create(path: str, is_file: bool = False, file_content: str = "") -> None:
+        if not os.path.exists(path):
+            if is_file:
+                with open(path, "w") as f:
```

Comment:
Using pathlib, things like this can be a little simpler:

```python
path = Path("df")
path.write_text(file_content.strip())
```


It will basically open up a context manager for you behind the scenes.